design/FY2025/Hybrid-DualFuel HVAC Controls.md

```diff
@@ -0,0 +1,387 @@
+  
+
+## Approach ##
+
+The proposed approach is to add options to the DemandManager system to lockout one fuel type or indicate a preference for one over another. The current demand manager objects act on specific load instances (e.g. Lights or Thermostats). From example file 5ZoneAirCooledDemandLimiting:
+```
+  DemandManager:ElectricEquipment,
+    Eq Mgr Stage 1,          !- Name
+    ,                        !- Availability Schedule Name
+    FIXED,                   !- Limit Control
+    60,                      !- Minimum Limit Duration {minutes}
+    0.0,                     !- Maximum Limit Fraction
+    ,                        !- Limit Step Change
+    ALL,                     !- Selection Control
+    ,                        !- Rotation Duration {minutes}
+    Space1-1 AllZones with Electric Equipment;  !- Electric Equipment 1 Name
+```
+
+Rather than target specific equipment objects, the fuel-switching demand manager will set a global status on each fuel type: `Available`, `Preferred`, or `NotAvailable`.
+Zone and airloop HVAC equipment will check the fuel-type status before activating the equipment. For plant equipment, the plant operation scheme will check the fuel-type status when dispatching supply equipment.
+
+An example `DemandManager:FuelSwitching` object:
+
+```
+DemandManager:FuelSwitching,
+  Dual Fuel Lowest Cost Lowest Carb, !- Name
+  ,                   !- Availability Schedule Name
+  Electricity,        !- Primary Fuel Type
+  NaturalGas,         !- Secondary Fuel Type
+  Heating,            !- End-Use Category
+  60,                 !- Minimum Duration {minutes}
+  None,               !- Capacity Control Type
+  ,                   !- Outdoor Dry Bulb Switchover Temperature {C}
+  None,               !- Above Switchover Temperature Control Type
+  None,               !- Below Switchover Temperature Control Type
+  Cost,               !- Criteria 1
+  PreferLowest,       !- Criteria 1 Control Type
+  CO2,                !- Criteria 2
+  PreferLowest;       !- Criteria 2 Control Type
+```
+
+In the example above, the following rules would be applied with a minimum of 60 minutes between status changes:
+
+* Capacity - Always use the primary fuel type (electricity) first, then supplement with the secondary fuel type.
+* CO2 - Use the fuel type with the lowest current CO2 emissions rate first given projected fuel consumption for the current timestep. Allow the secondary fuel type to be used if needed to supplement.
+
+So, what happens when different criteria favor different fuel types? The criteria will be applied in sequential fashion, similar to availability managers. When the demand manager is evaluated:
+
+1. Set all fuel types to `Available`.
```

Comment:
Use a schedule to make fuel types available? Would you then have to check these schedules to see if equipment were available all year or would that matter?